bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/discovery/AbstractMQTTDiscovery.java

```diff
@@ -48,11 +48,34 @@
 
     private @Nullable ScheduledFuture<?> scheduledStop;
 
+    private boolean isSubscribed;
```

Comment:
To make thsi really thread safe, this should be an `AtomicBoolean`.
Then can use `getAndSet()` to set and test the flag in a thread safe way.

Reply:
Ok.